### PR TITLE
ENT-3657: Ability to Revoke Code, but Not Email

### PR DESCRIPTION
**Pull request by saleem-latif**
JIRA Ticket: [ENT-3657](https://openedx.atlassian.net/browse/ENT-3657)

__Description:__
"As an admin, I want the ability to revoke my learner's code, without having to email them, to make the revoke a softer experience."

A/C:

1. Be able to revoke codes without having to email learners and notify them.

### CODE REVIEW

**Review by sameenfatima78 — APPROVED** [review]

File: ecommerce/extensions/api/serializers.py [hunk 1]
```diff
@@ -1769,6 +1770,8 @@ def create(self, validated_data):
         offer_assignments = validated_data.get('offer_assignments')
         email = validated_data.get('email')
         code = validated_data.get('code')
+        should_send_revoke_email = not validated_data.get('do_not_email')
```
Comment (zamanafzal) [inline ecommerce/extensions/api/serializers.py:1776]: Why are we using not here?

File: ecommerce/extensions/api/serializers.py [hunk 2]
```diff
@@ -1761,6 +1761,7 @@ class Meta:
     code = serializers.CharField(required=True)
     email = serializers.EmailField(required=True)
     detail = serializers.CharField(read_only=True)
+    do_not_email = serializers.BooleanField(default=False)
```
Comment (zamanafzal) [inline ecommerce/extensions/api/serializers.py:1764]: Can we choose a better name here? like revoke_without_email.

File: ecommerce/extensions/api/serializers.py (hunk 1 above)
Comment (saleem-latif) [reply ecommerce/extensions/api/serializers.py:1776]: just for a more descriptive variable name. this new variable can be used directly in the if statement like 
```
if should_send_revoke_email: 
    ...
```

File: ecommerce/extensions/api/serializers.py (hunk 2 above)
Comment (saleem-latif) [reply ecommerce/extensions/api/serializers.py:1764]: actually `Do Not Email` is the text provided by the UX and this is what is displayed on the frontend. That is why I used `do_not_email` here as well.

File: ecommerce/extensions/api/serializers.py (hunk 1 above)
Comment (zamanafzal) [reply ecommerce/extensions/api/serializers.py:1776]: Please add a comment for this as it confuses you when you see the code.

**Review by zamanafzal — APPROVED** [review]

**Pull request merged**